### PR TITLE
[Workaround] doctor xorm.Count nil on sqlite error

### LINKED ISSUE #11734: gitea doctor: panic: reflect: NumField of non-struct type
- Gitea version (or commit ref): 1.11.6
- Git version: 2.20.1
- Operating system: Linux Debian 10.4 (Buster)
- Database (use `[x]`):
  - [ ] PostgreSQL
  - [ ] MySQL
  - [ ] MSSQL
  - [x] SQLite
- Can you reproduce the bug at https://try.gitea.io:
  - [ ] Yes (provide example URL)
  - [ ] No
  - [x] Not relevant
- Log gist: 

## Description

Running gitea doctor results in a panic, see below.

I haven't noticed any issues running gitea.

```
git@netcup:~/gitea$ ./gitea doctor all
[ 1 ] Check paths and basic configuration
- Configuration File Path:   '/home/git/gitea/custom/conf/app.ini'
- Repository Root Path:      '/home/git/gitea-repositories'
- Data Root Path:            '/home/git/gitea/data'
- Custom File Root Path:     '/home/git/gitea/custom'
- Work directory:            '/home/git/gitea'
- Log Root Path:             '/home/git/gitea/log'
OK.

[ 2 ] Check Database Version
OK.

[ 3 ] Check if OpenSSH authorized_keys file is up-to-date
OK.

[ 4 ] Check consistency of database
panic: reflect: NumField of non-struct type

goroutine 1 [running]:
reflect.(*rtype).NumField(0x1b6ea20, 0x1b6ea20)
	/usr/local/go/src/reflect/type.go:981 +0x5e
code.gitea.io/gitea/vendor/xorm.io/xorm.(*Engine).mapType(0xc000144b60, 0x1b6ea20, 0x32e2b40, 0x98, 0x1b6ea00, 0x8, 0x18)
	/go/src/code.gitea.io/gitea/vendor/xorm.io/xorm/engine.go:929 +0x2a4
code.gitea.io/gitea/vendor/xorm.io/xorm.(*Engine).autoMapType(0xc000144b60, 0x1b6ea20, 0x32e2b40, 0x98, 0x0, 0x0, 0x0)
	/go/src/code.gitea.io/gitea/vendor/xorm.io/xorm/engine.go:855 +0x193
code.gitea.io/gitea/vendor/xorm.io/xorm.(*Statement).setRefBean(0xc001a8ee18, 0x1b6ea20, 0x32e2b40, 0x3349960, 0x0)
	/go/src/code.gitea.io/gitea/vendor/xorm.io/xorm/statement.go:222 +0x69
code.gitea.io/gitea/vendor/xorm.io/xorm.(*Statement).genCountSQL(0xc001a8ee18, 0xc000b832d0, 0x1, 0x1, 0x1e978a7, 0xc, 0xc000b83190, 0xb93b59, 0xc001ba51a0, 0x1b6ea20, ...)
	/go/src/code.gitea.io/gitea/vendor/xorm.io/xorm/statement.go:992 +0x3f0
code.gitea.io/gitea/vendor/xorm.io/xorm.(*Session).Count(0xc001a8ee00, 0xc000b832d0, 0x1, 0x1, 0x0, 0x0, 0x0)
	/go/src/code.gitea.io/gitea/vendor/xorm.io/xorm/session_stats.go:24 +0x259
code.gitea.io/gitea/cmd.runDoctorCheckDBConsistency(0xc0000c7340, 0x0, 0x0, 0x0, 0x0, 0x0)
	/go/src/code.gitea.io/gitea/cmd/doctor.go:521 +0x26b
code.gitea.io/gitea/cmd.runDoctor(0xc0000c7340, 0x0, 0xc0000c71e0)
	/go/src/code.gitea.io/gitea/cmd/doctor.go:210 +0x9ca
code.gitea.io/gitea/vendor/github.com/urfave/cli.HandleAction(0x1bc5ee0, 0x2ea6b88, 0xc0000c7340, 0xc0003bd700, 0x0)
	/go/src/code.gitea.io/gitea/vendor/github.com/urfave/cli/app.go:490 +0xc8
code.gitea.io/gitea/vendor/github.com/urfave/cli.Command.Run(0x1e85db0, 0x6, 0x0, 0x0, 0x0, 0x0, 0x0, 0x1ea4b37, 0x11, 0x0, ...)
	/go/src/code.gitea.io/gitea/vendor/github.com/urfave/cli/command.go:210 +0x991
code.gitea.io/gitea/vendor/github.com/urfave/cli.(*App).Run(0xc00009b1e0, 0xc00003c180, 0x3, 0x3, 0x0, 0x0)
	/go/src/code.gitea.io/gitea/vendor/github.com/urfave/cli/app.go:255 +0x6ab
main.main()
	/go/src/code.gitea.io/gitea/main.go:110 +0x845
git@netcup:~/gitea$
```

### PR DESCRIPTION
**Pull request by 6543**
since I dont think xorm will be updated in v1.11 branch - fix it with an workaround

fix #11734

### CODE REVIEW

**Review by zeripath — APPROVED** [review]

**Review by techknowlogick — APPROVED** [review]

**Comment by b-ehlers** [comment]
This PR fixes the panic, but creates a new issue "Error: ambiguous column name: id".
For details see @Dessa and my comment in #11734

File: models/consistency.go [hunk 1]
```diff
@@ -167,3 +168,23 @@ func (action *Action) checkForConsistency(t *testing.T) {
...
+		Select("id").Find(&ids)
+}
+
+// DeleteOrphanedObjects delete subjects with have no existing refobject anymore
+func DeleteOrphanedObjects(subject, refobject, joinCond string) error {
+	_, err := x.In("id", builder.Select("`"+subject+"`.id").
+		From("`"+subject+"`").
+		Join("LEFT", "`"+refobject+"`", joinCond).
+		Where(builder.IsNull{"`" + refobject + "`.id"})).
+		Delete("`" + subject + "`")
```
Comment (lunny) [inline models/consistency.go:188]: This is not supported by `xorm`. `Delete` needs an point but not a string.

**Review by lunny — COMMENTED** [review]
The bug resulted in all places which invoke `DeleteOrphanedObjects`.

**Comment by b-ehlers** [comment]
`Delete` may have another issue, but in my case the table  `tracked_time` is empty, so there should be no deletion.

I assume, that the select triggers my bug, the `Select("id")` uses the column 'id', which is ambiguous. Both `tracked_time` and the left joined `issue` contain the column 'id'.

```
+	return int64(len(ids)), x.Table("`"+subject+"`").
+		Join("LEFT", refobject, joinCond).
+		Where(builder.IsNull{"`" + refobject + "`.id"}).
+		Select("id").Find(&ids)
```

**Pull request merged**